llvm/include/llvm/Object/DXContainer.h

```diff
@@ -135,7 +188,36 @@ class RootSignature {
   uint32_t getRootParametersOffset() const { return RootParametersOffset; }
   uint32_t getNumStaticSamplers() const { return NumStaticSamplers; }
   uint32_t getStaticSamplersOffset() const { return StaticSamplersOffset; }
+  llvm::iterator_range<param_header_iterator> param_header() const {
+    return llvm::make_range(ParametersHeaders.begin(), ParametersHeaders.end());
+  }
+  param_header_iterator param_header_begin() {
+    return ParametersHeaders.begin();
+  }
+  param_header_iterator param_header_end() { return ParametersHeaders.end(); }
   uint32_t getFlags() const { return Flags; }
+
+  llvm::Expected<RootParameterView>
+  getParameter(const dxbc::RootParameterHeader &Header) const {
+    size_t CorrectOffset = Header.ParameterOffset - ParameterSpaceOffset;
+    StringRef Data;
+
```

Comment:
nit: there's a lot of excess whitespace around. Whitespace is good for breaking things up, but too much just makes it harder to read the code.
```suggestion
```